lang/src/org/partiql/lang/ast/ast.kt

```diff
@@ -964,6 +967,16 @@ enum class OrderingSpec {
     DESC
 }
 
+/**
+ * The sealed class includes all the datetime types such as DATE, TIME, TIMESTAMP
+ * Note: TIME and TIMESTAMP are yet to be added.
+ */
+sealed class DateTimeType : ExprNode() {
+    data class Date( val dateString: String, override val metas: MetaContainer ) : DateTimeType() {
+        override val children: List<AstNode> = listOf()
+    }
+}
```

Comment:
How should one think about this node relative to an Ion literal node that happens to be a timestamp?  I think we should at least have a comment about that.  Is the expectation that we normalize one form, or does the code have to deal with both?

Reply:
If this is modeled as an algebraic type, we should also consider making the AST form not have a `String` as the representation.  More specifically, why not model the `DATE` fields explicitly and require the parser to fill those in?

Reply:
This is different from the `Literal` AST node which corresponds to the Ion literals including `timestamp`. I will be introducing an AST node `TIMESTAMP` similar to `DATE` and `TIME`. The code will have to deal with both I guess. One would correspond to the Ion literal with the type `Timestamp` and one would be explicit `TIMESTAMP` ast node. 
I will add comments to mention this explicit difference.

Reply:
I understand that we aren't adding `TIMESTAMP` yet, but you allude to it, and I would like to understand how we explicitly want the code to operate.  It might be worth mentioning that this is orthogonal to the Ion literal nodes.

Reply:
The new `TIMESTAMP` node would be orthogonal to the `Ion literal` node. I have added a comment to mention this explicitly. 
The `TIMESTAMP` node will be instantiated for the timestamp literal `TIMESTAMP '2012-12-02 05:30:00'`, for example.
\`2012-12-02T05:30:00Z\` will correspond to the ion literal with the type `SqlDataType.TIMESTAMP`.
Does that answer the question?